src/timesync/timesyncd-manager.c

```diff
@@ -588,7 +588,7 @@ static int manager_receive_response(sd_event_source *source, int fd, uint32_t re
         }
 
         root_distance = ntp_ts_short_to_d(&ntpmsg.root_delay) / 2 + ntp_ts_short_to_d(&ntpmsg.root_dispersion);
-        if (root_distance > NTP_MAX_ROOT_DISTANCE) {
+        if ((usec_t) (root_distance * USEC_PER_SEC) > m->max_root_distance_usec) {
```

Comment:
hmm, given that "double" is kinda the "bigger" type than "usec_t", I'd probably cast `(double) m->max_root_distance_usec` here, rathern than the result of `root_distance` if you follow what i mean?

Reply:
OK. Updated.